### PR TITLE
Decaffeination

### PR DESCRIPTION
**Pull request by Adsidera**
Decaffeination of .coffee files present in src directory

### CODE REVIEW

**Comment by Adsidera** [comment]
Decaffeination went smoothly through the decaffeinate tool, except for glcui.coffee and huviz.coffee
that threw respectively the following errors:
image

image

**Comment by smurp** [comment]
https://coffeescript.org/#existential-operator

A variant use of the existential operator is to bail out of a property chain gracefully, as in the final example in the above section.

```coffee
zip = lottery.drawWinner?().address?.zipcode
```

becomes

```js
zip = typeof lottery.drawWinner === "function" ? (ref = lottery.drawWinner().address) != null ? ref.zipcode : void 0 : void 0;

```


It seems like bulk decaffeinate does not know about this syntax.

So in that `huviz.coffee` case:

```coffee
  pred_name = @ontology?label[pred_id]
```

we could do something like:

```coffee
    if @ontology?label
        pred_name = @ontology.label[pred_lid]
    @fire_newpredicate_event(pid, pred_lid, parent_lid, pred_name)
```

I think it is OK for us to call `@fire_newpredicate_event()` even when pred_name is undefined.

I will take a look at `gclui.coffee` before tomorrow morning.

**Comment by smurp** [comment]
I fixed the `gclui.coffee` issue.  It was a bug in `decaffeinate` concerning the handling of (at least) `on` as a method name in some circumstances. 

Here is the decaffeinate bug report I submitted...

https://github.com/decaffeinate/decaffeinate/issues/1439

File: src/huviz.coffee [hunk 1]
```diff
@@ -2861,7 +2861,8 @@ class Huviz
         parent_lid = "anything"
       @my_graph.predicates[pred_lid] = []
       @ensure_predicate_lineage(parent_lid)
-      pred_name = @ontology?label[pred_lid]
+      if @ontology.label
```
Comment (Adsidera) [inline src/huviz.coffee]: @smurp please check if the `if` statement on line 2684 is enough

Comment (Adsidera) [reply src/huviz.coffee]: thanks @smurp !

**Comment by Adsidera** [comment]
@smurp proceeding in the decaffeination of `huviz.coffee`, I had to refactor line 6155. 
Unfortunately, there is still the Decaffeination bug `UnhandledPromiseRejectionWarning: Error: unexpected ')'` appearing - see screenshot
image

**Pull request merged**